ts/screens/authentication/cie/CiePinScreen.tsx

```diff
@@ -48,140 +56,163 @@ const styles = StyleSheet.create({
+      <Link onPress={onOpenForgotPINPage}>
+        {I18n.t("authentication.cie.pin.bottomSheetCTA")}
+      </Link>
+    </View>,
+    I18n.t("authentication.cie.pin.bottomSheetTitle"),
+    300
+  );
+
+  const getContextualHelp = () => ({
     title: I18n.t("authentication.cie.pin.contextualHelpTitle"),
     body: () => (
       <Markdown>{I18n.t("authentication.cie.pin.contextualHelpBody")}</Markdown>
     )
   });
 
-  private onProceedToCardReaderScreen = (url: string) => {
-    const ciePin = this.state.pin;
-    this.setState({ url, pin: "" }, () => {
-      this.props.navigation.navigate({
-        routeName: ROUTES.CIE_CARD_READER_SCREEN,
-        params: { ciePin, authorizationUri: url }
-      });
-      this.props.hideModal();
-    });
+  const [pin, setPin] = useState("");
+  const [url, setUrl] = useState<string | undefined>(undefined);
+
+  const continueButtonRef = useRef<FooterWithButtons>(null);
+  const PINPadViewRef = useRef<View>(null);
+
+  const onProceedToCardReaderScreen = async (url: string) => {
+    setPin("");
+    setUrl(url);
   };
 
-  private handleAuthenticationOverlayOnClose = () => {
-    // reset the pin if user abort process during loading
-    this.setState({ pin: "" }, this.props.hideModal);
+  // Handles URL receival.
+  useEffect(() => {
+    if (url === undefined) {
+      return;
+    }
+
+    navigate({
+      routeName: ROUTES.CIE_CARD_READER_SCREEN,
+      params: {
+        ciePin: pin,
+        authorizationUri: url
+      }
+    });
+
+    hideModal();
+  }, [url]);
+
+  const handleAuthenticationOverlayOnClose = () => {
+    setPin("");
+    hideModal();
   };
 
-  private showModal = () => {
-    this.props.requestNfcEnabledCheck();
+  const showModal = () => {
+    props.requestNfcEnabledCheck();
+
     Keyboard.dismiss();
-    const component = (
+
+    showAnimatedModal(
       <CieRequestAuthenticationOverlay
-        onClose={this.handleAuthenticationOverlayOnClose}
-        onSuccess={this.onProceedToCardReaderScreen}
-      />
+        onClose={handleAuthenticationOverlayOnClose}
+        onSuccess={onProceedToCardReaderScreen}
+      />,
+      BottomTopAnimation
     );
-    this.props.showAnimatedModal(component, BottomTopAnimation);
   };
 
-  // Method called when the PIN changes
-  public handelOnPinChanged = (pin: string) => {
-    this.setState(
-      {
-        pin
-      },
-      () => {
-        // set focus on continue button when the pin input is full filled
-        if (this.state.pin.length === CIE_PIN_LENGTH) {
-          setAccessibilityFocus(this.continueButtonRef, 100 as Millisecond);
-        }
-      }
-    );
-  };
+  const onPinChanged = (pin: string) => setPin(pin);
+
+  useEffect(() => {
+    if (pin === "") {
+      return;
+    }
+
+    if (pin.length === CIE_PIN_LENGTH) {
+      setAccessibilityFocus(continueButtonRef, 100 as Millisecond);
+    }
+  }, [pin]);
+
+  const doSetAccessibilityFocus = useCallback(() => {
+    if (!PINPadViewRef.current) {
```

Comment:
```suggestion
```
`setAccessibilityFocus` already does the check about nullish on `.current`
Indeed this check could be avoided